client/src/components/AuditAdmin/Setup/Settings/Settings.tsx

```diff
@@ -15,6 +23,70 @@ import {
 } from '../../../useAuditSettings'
 import { stateOptions } from './states'
 import { range } from '../../../../utils/array'
+import AsyncButton from '../../../Atoms/AsyncButton'
+
+const RANDOM_SEED_INSTRUCTIONS =
+  'Enter a series of random numbers to be used when drawing the random sample of ballots to audit.'
+
+const RandomSeedModal = ({
+  initialValue,
+  onClose,
+  onSave,
+}: {
+  initialValue: string | null
+  onClose: () => void
+  onSave: (seed: string) => Promise<void>
+}): JSX.Element => {
+  const [value, setValue] = React.useState(initialValue || '')
+  return (
+    <Dialog
+      isOpen
+      onClose={onClose}
+      title="Enter Random Seed"
+      // Zoom in the whole modal
+      style={{ transform: 'scale(2)' }}
+      // Darken the usual backdrop
+      backdropProps={{ style: { backgroundColor: 'rgba(16, 22, 26, 0.9)' } }}
+    >
+      <div className={Classes.DIALOG_BODY}>
+        <label htmlFor="random-seed-modal" id="random-seed-modal-label">
+          <p>{RANDOM_SEED_INSTRUCTIONS}</p>
+          <InputGroup
+            id="random-seed-modal"
+            aria-labelledby="random-seed-modal-label"
+            type="text"
+            value={value}
+            onChange={(e: React.FormEvent<HTMLInputElement>) =>
+              setValue(e.currentTarget.value)
+            }
+            large
+            fill
+            style={{
+              fontSize: '1.5em',
+              letterSpacing: '0.2em',
+            }}
+          />
+        </label>
+      </div>
+      <div className={Classes.DIALOG_FOOTER}>
+        <div className={Classes.DIALOG_FOOTER_ACTIONS}>
+          <Button onClick={onClose}>Cancel</Button>
+          <AsyncButton
+            intent="success"
+            icon="tick"
+            disabled={!value}
+            onClick={async () => {
+              await onSave(value)
```

Comment:
Intentional to not have any input validation here? Wondering what would happen if someone input "   "

Reply:
I don't think there was any validation in place here. This form is quite old (before my time) and could probably use some attention in that regard. There is some server-side validation to ensure it's a string, but no length requirements. Technically, Arlo can use any string as a random seed, so there's not a high risk, but it might be worth thinking through if there's any validation that would help users. Due to time constraints I don't want to block this PR on this, but happy to discuss more and see if we think there's a follow up that would make sense.